### PR TITLE
Added `openFiltersMode` on `filter-navigator`

### PR DESCRIPTION
**Pull request by iaronaraujo**
#### What problem is this solving?

Add a prop to make it possible to only have one AvailableFilter open at the same time. Solves this issue https://github.com/vtex-apps/search-result/issues/381 and is based on this PR https://github.com/vtex-apps/search-result/pull/397
The credit of this PR should go to @felipeireslan since he created the PR this was based on.

#### How to test it?

[Workspace](https://iaronaraujo--storecomponents.myvtex.com/apparel---accessories/clothing/)

#### Screenshots or example usage:

Sep-04-2020 14-00-24

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -286,6 +286,7 @@ Check out the [**Product Summary documentation**](https://vtex.io/docs/component
 | `maxItemsDepartment` | `number`                 | Maximum number of departments to be displayed before the See More button is triggered.          | `8`             |
 | `maxItemsCategory`   | `number`                 | Maximum number of category items to be displayed before the See More button is triggered.     | `8`             |
 | `initiallyCollapsed` | `Boolean` | Makes the search filters start out collapsed (`true`) or open (`false`). | `false` |
+| `openFiltersMode`    | `Enum` | Dictates how many filters can be open at the same time. It is possible to have many open at once (`MANY`) or just one (`ONE`). Notice that if `ONE` is selected, the filters will begin collapsed, regardless of what is passed to `initiallyCollapsed` | `MANY` |
```
Comment (klzns) [inline docs/README.md:289]:
```suggestion
| `openFiltersMode`    | `Enum` | Dictates how many filters can be open at the same time. It is possible to have many open at once (`many`) or just one (`one`). Notice that if `one` is selected, the filters will begin collapsed, regardless of what is passed to `initiallyCollapsed` | `many` |
```

File: react/FilterNavigator.js [hunk 2]
```diff
@@ -72,6 +72,7 @@ const FilterNavigator = ({
   maxItemsCategory = 8,
   filtersTitleHtmlTag = 'h5',
   scrollToTop = 'none',
+  openFiltersMode = 'MANY',
```
Comment (klzns) [inline react/FilterNavigator.js:75]:
```suggestion
  openFiltersMode = 'many',
```

File: react/components/AvailableFilters.js [hunk 3]
```diff
@@ -10,8 +10,10 @@ const AvailableFilters = ({
   preventRouteChange = false,
   initiallyCollapsed = false,
   navigateToFacet,
-}) =>
-  filters.map(filter => {
+  openFiltersMode = 'MANY',
```
Comment (klzns) [inline react/components/AvailableFilters.js:13]:
```suggestion
  openFiltersMode = 'many',
```

File: react/components/FilterOptionTemplate.js [hunk 4]
```diff
@@ -48,6 +51,7 @@ const FilterOptionTemplate = ({
       filter => filter.name.toLowerCase().indexOf(searchTerm.toLowerCase()) > -1
     )
   }, [filters, searchTerm, thresholdForFacetSearch])
+  const isOpen = openFiltersMode === 'MANY' ? open : lastOpenFilter === title
```
Comment (klzns) [inline react/components/FilterOptionTemplate.js:54]:
```suggestion
  const isOpen = openFiltersMode === 'many' ? open : lastOpenFilter === title
```

File: react/components/FilterOptionTemplate.js [hunk 5]
```diff
@@ -57,14 +61,26 @@ const FilterOptionTemplate = ({
...
+  const handleCollapse = useCallback(() => {
+    if (openFiltersMode === 'MANY') {
+      setOpen(!open)
+    } else if (openFiltersMode === 'ONE') {
+      setLastOpenFilter(lastOpenFilter === title ? null : title)
+    } else {
+      console.error(
+        `Invalid openFiltersMode value: ${openFiltersMode}\nCheck the documentation for the values available`
+      )
+    }
```
Comment (klzns) [inline react/components/FilterOptionTemplate.js:73]:
```suggestion
    if (openFiltersMode === 'many') {
      setOpen(!open)
      return
    }
    if (openFiltersMode === 'one') {
      setLastOpenFilter(lastOpenFilter === title ? null : title)
      return
    }
    
    console.error(
        `Invalid openFiltersMode value: ${openFiltersMode}\nCheck the documentation for the values available`
      )
```

**Review by klzns — CHANGES_REQUESTED** [review]

File: docs/README.md [hunk 6]
```diff
@@ -286,6 +286,7 @@ Check out the [**Product Summary documentation**](https://vtex.io/docs/component
 | `maxItemsDepartment` | `number`                 | Maximum number of departments to be displayed before the See More button is triggered.          | `8`             |
 | `maxItemsCategory`   | `number`                 | Maximum number of category items to be displayed before the See More button is triggered.     | `8`             |
 | `initiallyCollapsed` | `Boolean` | Makes the search filters start out collapsed (`true`) or open (`false`). | `false` |
+| `openFiltersMode`    | `Enum` | Dictates how many filters can be open at the same time. It is possible to have many open at once (`many`) or just one (`one`). Notice that if `one` is selected, the filters will begin collapsed, regardless of what is passed to `initiallyCollapsed` | `many` |
```
Comment (guerreirobeatriz) [inline docs/README.md:289]:
```suggestion
| `openFiltersMode`    | `Enum` | Defines how many filter categories can be opened at the same time on the Filter Navigator component. Possible values are: `many` (more than one filter category can be opened at the same time) and `one` (only one filter category can be opened). Notice that if `one` is declared, the filters will always be collapsed before user interaction, regardless of what is passed to the `initiallyCollapsed` prop. | `many` |
```

**Review by guerreirobeatriz — CHANGES_REQUESTED** [review]

File: react/components/AvailableFilters.js [hunk 7]
```diff
@@ -23,7 +23,9 @@ const Filter = ({
   initiallyCollapsed,
   navigateToFacet,
   lazyRender,
+  openFiltersMode = 'many',
```
Comment (victorhmp) [inline react/components/AvailableFilters.js:26]: Just a suggestion, but how about calling this mode `multiple` instead of `many`?

**Review by victorhmp — APPROVED** [review]

**Comment by ricardo7634234** [comment]
Good morning guys, do we have an update? What´s next?

**Comment by iaronaraujo** [comment]
Hi @ricardo7634234 !  After some of the most recent changes to this app, this solution stopped working, so I am still investigating how to make this work properly!

**Review by klzns — APPROVED** [review]

**Review by guerreirobeatriz — APPROVED** [review]

**Pull request merged**